pkg/repository/metered/metered.go

```diff
@@ -30,6 +30,69 @@ func NewMetered(entitlements repository.EntitlementsRepository, l *zerolog.Logge
 	}
 }
 
+func (m *Metered) canCreate(ctx context.Context, resource dbsqlc.LimitResource, tenantId string, numberOfResources int32) (bool, int, error) {
+	var key = fmt.Sprintf("%s:%s", resource, tenantId)
+
+	var canCreate *bool
+	var percent int
+
+	if hit, ok := m.c.Get(key); ok {
+		c := hit.(bool)
+		canCreate = &c
+	}
+
+	if canCreate == nil {
+		c, p, err := m.entitlements.TenantLimit().CanCreate(ctx, resource, tenantId, numberOfResources)
+
+		if err != nil {
+			return false, 0, fmt.Errorf("could not check tenant limit: %w", err)
+		}
+
+		canCreate = &c
+		percent = p
+
+		if percent <= 50 || percent >= 100 {
+			m.c.Set(key, c)
+		}
+	}
```

Comment:
not sure I'm following this - what are the percents here? usage?

Reply:
yeah we don't cache as we get close to the limit -- this is old code that was copy-pasted, we might want to change this...